### PR TITLE
Adds support for Green Aura MVP damage

### LINKED ISSUE #3324: Boss Damage Update
* **rAthena Hash**: Any
* **Client Date**: Any
* **Server Mode**: RE
* **Description of Issue**: Now in the kRO the mvp's only receive 10% of the player's damage.
  * Official Information: http://ro.gnjoy.com/news/devnote/View.asp?category=2&seq=3891525&curpage=1

https://media.giphy.com/media/7zSzFATT4pVbJFpv6M/giphy.gif

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: Fixes #3324

* **Server Mode**: Renewal

* **Description of Pull Request**: 
  * Adds the DamageTaken flag which reduces damage done MVP.
Thanks to @admkakaroto and @SonaSonaSona!

### CODE REVIEW

File: db/import-tmpl/mob_db.yml [hunk 1]
```diff
@@ -55,6 +55,7 @@
 #   AttackDelay             Attack speed. (Default: 0)
 #   AttackMotion            Attack animation speed. (Default: 0)
 #   DamageMotion            Damage animation speed. (Default: 0)
+#   DamageTaken             Rate at which the monster will recieve damage from players. (Default: 100)
```
Comment (Atemo) [inline db/import-tmpl/mob_db.yml:58]:
```suggestion
#   DamageTaken             Rate at which the monster will receive damage from players. (Default: 100)
```

File: db/mob_db.yml [hunk 2]
```diff
@@ -55,6 +55,7 @@
 #   AttackDelay             Attack speed. (Default: 0)
 #   AttackMotion            Attack animation speed. (Default: 0)
 #   DamageMotion            Damage animation speed. (Default: 0)
+#   DamageTaken             Rate at which the monster will recieve damage from players. (Default: 100)
```
Comment (Atemo) [inline db/mob_db.yml:58]:
```suggestion
#   DamageTaken             Rate at which the monster will receive damage from players. (Default: 100)
```

File: db/pre-re/mob_db.yml [hunk 3]
```diff
@@ -55,6 +55,7 @@
 #   AttackDelay             Attack speed. (Default: 0)
 #   AttackMotion            Attack animation speed. (Default: 0)
 #   DamageMotion            Damage animation speed. (Default: 0)
+#   DamageTaken             Rate at which the monster will recieve damage from players. (Default: 100)
```
Comment (Atemo) [inline db/pre-re/mob_db.yml:58]:
```suggestion
#   DamageTaken             Rate at which the monster will receive damage from players. (Default: 100)
```

File: db/re/mob_db.yml [hunk 4]
```diff
@@ -55,6 +55,7 @@
 #   AttackDelay             Attack speed. (Default: 0)
 #   AttackMotion            Attack animation speed. (Default: 0)
 #   DamageMotion            Damage animation speed. (Default: 0)
+#   DamageTaken             Rate at which the monster will recieve damage from players. (Default: 100)
```
Comment (Atemo) [inline db/re/mob_db.yml:58]:
```suggestion
#   DamageTaken             Rate at which the monster will receive damage from players. (Default: 100)
```

File: doc/mob_db.txt [hunk 5]
```diff
@@ -200,6 +200,10 @@ DamageMotion: Damage animation motion, same as aMotion but used to display the "
 
 ---------------------------------------
 
+DamageTaken: Rate at which the monster will recieve damage from players.
```
Comment (Atemo) [inline doc/mob_db.txt:203]:
```suggestion
DamageTaken: Rate at which the monster will receive damage from players.
```

File: doc/yaml/db/mob_db.yml [hunk 6]
```diff
@@ -38,6 +38,7 @@
 #   AttackDelay             Attack speed. (Default: 0)
 #   AttackMotion            Attack animation speed. (Default: 0)
 #   DamageMotion            Damage animation speed. (Default: 0)
+#   DamageTaken             Rate at which the monster will recieve damage from players. (Default: 100)
```
Comment (Atemo) [inline doc/yaml/db/mob_db.yml:41]:
```suggestion
#   DamageTaken             Rate at which the monster will receive damage from players. (Default: 100)
```

File: src/map/battle.cpp [hunk 7]
```diff
@@ -1797,6 +1797,13 @@ int64 battle_calc_damage(struct block_list *src,struct block_list *bl,struct Dam
 		pc_overheat(tsd, (element == ELE_FIRE ? 3 : 1));
 	}
 
+	if (bl->type == BL_MOB) { // Reduces damage received for naturally spawned Green Aura MVP
+		mob_data *md = BL_CAST(BL_MOB, bl);
+
+		if (md && md->spawn_timer != INVALID_TIMER && md->state.boss && md->db->damagetaken != 100)
+			damage = i64max(damage * (100 - md->db->damagetaken) / 100, 1);
```
Comment (Atemo) [inline src/map/battle.cpp:1804]:
```suggestion
			damage = i64max(damage * md->db->damagetaken / 100, 1);
```

File: src/map/battle.cpp [hunk 8]
```diff
@@ -1797,6 +1797,13 @@ int64 battle_calc_damage(struct block_list *src,struct block_list *bl,struct Dam
...
+	}
```
Comment (Atemo) [inline src/map/battle.cpp:1805]: `Rate at which the monster will receive damage from players. (Default: 100)` there is no tsd check, so the damage is reduced from others source too?

**Comment by vstumpf** [comment]
Do we really need to bump the version? The addition of the new key isn't a breaking change, unless you're worried for someone using a newer db.

**Review by vstumpf — APPROVED** [review]

**Pull request merged**